File: 프로그래머스/lv1/이선영/시저 암호.java
import java.util.*;
class Solution {
    public String solution(String s, int n) {
        StringBuilder sb = new StringBuilder();
   
        for(int i=0; i<s.length(); i++){
            char c = s.charAt(i);
            char next = (char)(c+n);
            
            if((int)c >=97 && (int)c<=122){    
                if((int)next > 122){
                    next = (char)((next%123)+97);
                   
                }
            }else if((int) c >=65 && (int) c<=90){
                if((int)next > 90){
                    next = (char)((next%91)+65);
                }
            }else if(c==' '){
                next = c;
            }
            sb.append(next);
        }
        
        return sb.toString();
    }
}

File: 프로그래머스/lv1/이하은/시저 암호.kt
/*
대문자: 65~90
소문자: 97~122
*/
class Solution {
    fun solution(s: String, n: Int): String {
        val sb = StringBuilder(s)
        
        for(i in 0 until s.length){
            val ch = s[i]
            if(ch == ' ') continue
            
            var start = 0
            if(ch.isUpperCase()){
                start = 65
            }
            
            if(ch.isLowerCase()){
                start = 97
            }
            
            val next = (ch.toInt() + n - start) % 26 + start
            sb.replace(i, i + 1, next.toChar().toString())
        }
        
        return sb.toString()
    }
}